Update Domain / In Domain and sub members

yoyo-go opened the pull request:
Domain / In Domain
Domain Topic / Has Domain Topic
Domain Region / Has Domain Region
Exemplifies / Is Exemplified By

fcbond commented:
This looks like a great start.

A couple of comments
 - If we are moving to using A and B (not X and Y), can you fix the xml snippet as well?
 - the examples seem to be reversed for exmplifies/is_exemplified_by
 - can you add a note to the top level domain (maybe a comment) that in EuroWordnet these were moved to a separate ontology (EWN guidelines pp 8-10).
 - The examples lack proper spacing: football''domain topic''place-kick
 - Instead of using quotes (like "relation") can we use some kind of style `<span class='sr'>domain topic</span>` and style them all uniformly?   I guess this should go into the contributing guidelines.
 - maybe a more common word than _phytology_ would be better for the example.

yoyo-go commented:
> This looks like a great start.
> 
> A couple of comments
> 
> * If we are moving to using A and B (not X and Y), can you fix the xml snippet as well?

Update the xml snippet and `templates/relation.html`, it's done.

> * the examples seem to be reversed for exmplifies/is_exemplified_by

Exmplifies/is_exemplified_by seem to be totally reversed at the first draft, hope I make everything right this time.

> * can you add a note to the top level domain (maybe a comment) that in EuroWordnet these were moved to a separate ontology (EWN guidelines pp 8-10).

Done

> * The examples lack proper spacing: football''domain topic''place-kick

Fixed

> * Instead of using quotes (like "relation") can we use some kind of style `<span class='sr'>domain topic</span>` and style them all uniformly?   I guess this should go into the contributing guidelines.

Do you mean the "relation" in the examples, or in other place?

> * maybe a more common word than _phytology_ would be better for the example.

Changed this to another example

PS I have changed all the examples into the new format with ILIID now, in order to change them into links.

fcbond commented:
Hi, 

when I try and build this, I get errors:
`$ python docs/build.py --lang en html > docs/index.en.html`
`<string>:1: (WARNING/2) Inline emphasis start-string without end-string.`

Please ensure that it build without errors before you commit.

I think that you cannot have the italics and hyperlink (which is a bit annoying).  I think we should lose the italics for now.

Further, you have `relations.domain.com.en` twice.   I think one should be `relations.has_domain.com.en`.  It is also weirdly formatted, but I think that may be connected to the other bug.

yoyo-go commented:
> Hi,
> 
> when I try and build this, I get errors:
> `$ python docs/build.py --lang en html > docs/index.en.html`
> `<string>:1: (WARNING/2) Inline emphasis start-string without end-string.`
> 
> Please ensure that it build without errors before you commit.
> 
> I think that you cannot have the italics and hyperlink (which is a bit annoying). I think we should lose the italics for now.
> 
> Further, you have `relations.domain.com.en` twice. I think one should be `relations.has_domain.com.en`. It is also weirdly formatted, but I think that may be connected to the other bug.

Hi Francis, I have tested and revised the code, but before I submit the commit, I would like to check with you about the look of the example:
![aaa](https://user-images.githubusercontent.com/56577574/95015975-a34a4200-0682-11eb-8c8a-321f7a0b03ff.PNG)
It now looks like this on my device, is this good for you this time? and we have two types on formatting examples, do we need to unify them? I think type A is clearer than type B, do we need  `` `` between the relation name as in type B, and add is XYZ of as type A seems to be more friendly with users.

Just want to make sure that is what you want, and I will submit the commit after your confirmation.

fcbond commented:
I think (and I think Michael also says elsewhere) that type A is better.

The contributing guide should specify it.


On Sun, Oct 4, 2020 at 8:55 PM yoyo-go <notifications@github.com> wrote:

> Hi,
>
> when I try and build this, I get errors:
> $ python docs/build.py --lang en html > docs/index.en.html
> <string>:1: (WARNING/2) Inline emphasis start-string without end-string.
>
> Please ensure that it build without errors before you commit.
>
> I think that you cannot have the italics and hyperlink (which is a bit
> annoying). I think we should lose the italics for now.
>
> Further, you have relations.domain.com.en twice. I think one should be
> relations.has_domain.com.en. It is also weirdly formatted, but I think
> that may be connected to the other bug.
>
> Hi Francis, I have tested and revised the code, but before I submit the
> commit, I would like to check with you about the look of the example:
> [image: aaa]
> <https://user-images.githubusercontent.com/56577574/95015975-a34a4200-0682-11eb-8c8a-321f7a0b03ff.PNG>
> It now looks like this on my device, is this good for you this time? and
> we have two types on formatting examples, do we need to unify them? I think
> type A is clearer than type B, do we need between the relation name as in
> type B, and add is XYZ of as type A seems to be more friendly with users.
>
> Just want to make sure that is what you want, and I will submit the commit
> after your confirmation.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/globalwordnet/gwadoc/pull/35#issuecomment-703251498>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIPZRRDJUIKSYIR5XGSA7DSJBWCFANCNFSM4R3TUTOA>
> .
>


-- 
Francis Bond <http://www3.ntu.edu.sg/home/fcbond/>
Division of Linguistics and Multilingual Studies
Nanyang Technological University

goodmami commented:
> I think (and I think Michael also says elsewhere) that type A is better.

Yes, but I also say it's better than some Type C "evergreen has domain-topic plant" which inverts the directionality.

We also cannot apply the "A is a (relation) of B" template too strictly, because some relations are not very nouny ("A is a see-also of B", "A is an exemplifies of B", etc... these sound weird), so our contributing guide should perhaps make recommendations for these cases.

fcbond commented:
I think the content is pretty much ready to go.

Because some of the files have changed places, I think it might be easiest if you checkout a current branch of gwadoc, then merge your changes with that and make a new pull request.  @goodmami do you have any better ideas?

goodmami commented:
> Because some of the files have changed places, I think it might be easiest if you checkout a current branch of gwadoc, then merge your changes with that and make a new pull request. @goodmami do you have any better ideas?

I think that's a good suggestion. There are ways to make git rebase to the new file (see [this StackOverflow answer](https://stackoverflow.com/a/15137607/1441112)), but I think it would be easier to just manually copy the changes into the new files using a merge tool. If you don't have a visual merge tool that you like already, I've been happy with [Meld](https://meldmerge.org/), and for Windows there's also [WinMerge](https://winmerge.org/). Some editors like Visual Studio Code and Emacs will have diff tools built-in, as well. **Don't** just overwrite the entire file with your copy, because there may be changes since you branched and they would get lost.

The steps are:
- Create copies of the 4 changed files
- Create a new branch off of the current `gsod` branch
- Merge your copied files into their new location
- Push the new branch

Or, as commands:

```bash
# copy changed files
cp docs/snippets.xml my-snippets.xml
cp docs/templates/relation.html my-relation.html
cp gwadoc/doc_basic.py my-doc_basic.py
cp gwadoc/doc_en.py my-doc_en.py
# create a new branch (first update gsod from upstream)
git checkout gsod  # start from this branch
git pull upstream/gsod  # assumes you have set globalwordnet/gwadoc as 'upstream'
git checkout -b Update-Domain-In-Domain-and-sub-members-2
# merge changes (here using Meld; substitute your tool if necessary)
meld my-snippets.xml templates/snippets.xml
meld my-relation.html templates/relation.html
meld my-doc_basic.py gwadoc/doc_basic.py
meld my-doc_en.py gwadoc/doc_en.py
# push the new branch
git commit
git push --set-upstream origin Update-Domain-In-Domain-and-sub-members-2
```

Then you can make a new PR. If you want to update the files of this PR instead, you can force-push to this branch. I think this would work:

```bash
git push --force origin Update-Domain-In-Domain-and-sub-members
```

Hope that works for you

fcbond commented:
This has been superseded by a new request.
